### PR TITLE
BREAKING: upgrade supabase to v1.0.0

### PR DESCRIPTION
**Pull request by dshukertjr**
- Upgrade supabase to v1.0.0
- Update the minimum supported Dart to 2.15.0 and  Flutter to 2.8.0
- Modify github actions to test the library against the above versions of Flutter

### CODE REVIEW

File: lib/src/supabase_auth.dart [hunk 1]
```diff
@@ -42,29 +42,13 @@ class SupabaseAuth with WidgetsBindingObserver {
   bool _initialDeeplinkIsHandled = false;
   String? _authCallbackUrlHostname;
 
-  GotrueSubscription? _authSubscription;
+  StreamSubscription<AuthState>? _authSubscription;
   final _listenerController = StreamController<AuthChangeEvent>.broadcast();
```
Comment (Vinzent03) [inline lib/src/supabase_auth.dart:46]: This can be removed

**Review by Vinzent03 — CHANGES_REQUESTED** [review]
I think the changelog is missing.

**Comment by dshukertjr** [comment]
Ah, I have deleted too much. Hold on.

**Comment by dshukertjr** [comment]
Should be fine now!

**Review by Vinzent03 — APPROVED** [review]

**Pull request merged**